src/data/fintype/fin.lean

```diff
@@ -45,4 +47,27 @@ begin
   { rintro ⟨i, hi, rfl⟩, simpa },
 end
 
+lemma card_filter_univ_fin (p : (fin (n + 1)) → Prop) [decidable_pred p] :
+  (univ.filter p).card = (ite (p 0) 1 0) + (univ.filter (p ∘ fin.succ)).card :=
```

Comment:
Can you not make progress much faster here with `rw fin.univ_fin_succ` or whatever the lemma is called?

Reply:
Yes that did make it extremely simpler